### ISSUE TITLE
`String.hex_to_int()` parse int32 instead of int64

### ISSUE THREAD

**Issue by zJudGenie** [godotengine/godot#56285]
### Godot version

3.4.2.stable

### System information

Windows 10

### Issue description

Hello, I was writing some code in GDScript to parse an UUID from string to two `int64` and I found that if you try to parse values bigger than `0x7fffffff`, using `hex_to_int()` it gives you

```cpp
ERROR: Cannot represent 0xb5d54688 as integer, provided value is too big.
    at: (core/ustring.cpp:1721)
```
theoretically, because the `int` type in GDScript is a signed 64-bit integer it should parse until `0xffff ffff ffff ffff` or at least `0x7fff ffff ffff ffff` am I wrong ?

### Steps to reproduce

```gdscript
var test: int = 0xb6d6fc4e04a9
print(test) # correct, prints 201034472228009

print("0x7fffffff".hex_to_int()) # correct, prints 2147483647
print("0xb5d54688".hex_to_int()) # wrong, prints 2147483647
```

**Comment by Calinou** [comment]
I can confirm this on Godot 3.3.4, so this is not a regression in 3.4:

![image](https://user-images.githubusercontent.com/180032/147499909-df8ee6f1-6452-4ea8-b53a-261d5b8fb8d1.png)

**Comment by Rubonnek** [comment]
`String::hex_to_int64` exists. I suppose it's a matter of exposing it over to GDScript.

https://github.com/godotengine/godot/blob/a75afd61a77c58fc7daaa525e485398795221f6f/core/ustring.cpp#L1761-L1801

**Comment by Calinou** [comment]
> `String::hex_to_int64` exists. I suppose it's a matter of exposing it over to GDScript.

Shouldn't we expose `hex_to_int64` as `hex_to_int` to replace the existing function? I'm not sure if anyone is currently relying on the overflowing behavior.

**Comment by zJudGenie** [comment]
> `String::hex_to_int64` exists. I suppose it's a matter of exposing it over to GDScript.
> 
> https://github.com/godotengine/godot/blob/a75afd61a77c58fc7daaa525e485398795221f6f/core/ustring.cpp#L1761-L1801

I am a little bit confused by this method, what is the one called by GDScript ?
https://github.com/godotengine/godot/blob/91c05290738aae5662a76f59dd3e4ccfdf5e37ae/core/string/ustring.cpp#L2121-L2160

**Comment by Rubonnek** [comment]
> Shouldn't we expose hex_to_int64 as hex_to_int to replace the existing function?

I agree. My only concern is backwards compatibility, but I don't think anyone is relying on the 32-bit clamping behavior either.

If we decide to do so, however, we might as well do it across the board in `3.x` for all the other `String` functions.

For example, `String.to_int` also uses 32-bit integers.

The following will also result in the same behavior:

```gdscript
print(str("3050653320").to_int())
```

There are other 64-bit variants that are not exposed such as `bin_to_int64`, and perhaps a few more I don't recall right now.

**Comment by Rubonnek** [comment]
> I am a little bit confused by this method, what is the one called by GDScript ?

@zJudGenie the second one you posted is from the `master` branch.

In `3.x`, the binding over to `GDScript` is happening in `core/variant_call.cpp`

Respectively at:
https://github.com/godotengine/godot/blob/a75afd61a77c58fc7daaa525e485398795221f6f/core/variant_call.cpp#L319

https://github.com/godotengine/godot/blob/a75afd61a77c58fc7daaa525e485398795221f6f/core/variant_call.cpp#L1722

**Comment by theoniko** [comment]
Hello all,
Could i work on this? Could you give some hint about the intended behavior? As far as i understood i just  need to refactor hex_to_int()

**Comment by akien-mga** [comment]
Fixed by #56363.
